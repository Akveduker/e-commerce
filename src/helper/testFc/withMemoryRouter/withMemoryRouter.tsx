import { render } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"

export const withMemoryRouter = (children: React.ReactNode) => {
    return render(
        <MemoryRouter>
            {children}
        </MemoryRouter>
    )
}