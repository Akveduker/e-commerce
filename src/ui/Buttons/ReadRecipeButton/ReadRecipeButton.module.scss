@import "../../../styles/_mixins.scss";
@import "../../../styles/_vars.scss";
.button {
    padding: 12px 8px 12px 16px;
    display: inline-flex;
    justify-content: center;
    align-items: center;
    @include fontBold;
    border-radius: $borderRadius;
    border: 2px solid $C2C;
    text-decoration: none;
    color: $C1A;
}
.icon {
    min-width: 16px;
    height: 16px;
    margin-left: 6px;
    box-sizing: content-box;
    path {
        stroke: $C2A;
    }
}
