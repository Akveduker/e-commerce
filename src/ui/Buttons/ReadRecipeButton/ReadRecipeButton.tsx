import React, { FC } from 'react';
import { Link, LinkProps } from 'react-router-dom';
import ArrowRight from '../../icons/ArrowRight/ArrowRight';
import s from './ReadRecipeButton.module.scss'
const ReadRecipeButton: FC<LinkProps> = (props) => {
    return (
        <Link className={s['button']} {...props}>
            Посмотреть рецепты
            <ArrowRight
                className={s['icon']}
            />
        </Link>
    );
};

export default ReadRecipeButton;