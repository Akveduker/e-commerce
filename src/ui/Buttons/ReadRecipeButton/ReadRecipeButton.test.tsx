import { withMemoryRouter } from "../../../helper/testFc/withMemoryRouter/withMemoryRouter"
import ReadRecipeButton from "./ReadRecipeButton"

describe('ReadRecipeButton', () => {
    test('Снапшот', () => {
        const { container } = withMemoryRouter(
            <ReadRecipeButton to={'/'} />
        )
        expect(container).toMatchSnapshot()
    })
})