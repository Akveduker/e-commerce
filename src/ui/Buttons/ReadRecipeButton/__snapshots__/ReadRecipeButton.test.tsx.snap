// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadRecipeButton Снапшот 1`] = `
<div>
  <a
    class="button"
    href="/"
  >
    Посмотреть рецепты
    <svg
      class="icon"
      fill="none"
      height="17"
      viewBox="0 0 17 17"
      width="17"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7.0332 11.6867L9.7732 8.94669C9.89737 8.82178 9.96706 8.65281 9.96706 8.47669C9.96706 8.30056 9.89737 8.13159 9.7732 8.00669L7.10654 5.34002"
        stroke="#151515"
        stroke-linecap="round"
        stroke-linejoin="bevel"
        stroke-width="2.5"
      />
    </svg>
  </a>
</div>
`;
